sqlx-cli/src/lib.rs

```diff
@@ -124,26 +125,21 @@ where
 
     let db_url = opts.required_db_url()?;
 
-    backoff::future::retry(
-        backoff::ExponentialBackoffBuilder::new()
-            .with_max_elapsed_time(Some(Duration::from_secs(opts.connect_timeout)))
-            .build(),
-        || {
-            connect(db_url).map_err(|e| -> backoff::Error<anyhow::Error> {
-                if let sqlx::Error::Io(ref ioe) = e {
-                    match ioe.kind() {
-                        io::ErrorKind::ConnectionRefused
-                        | io::ErrorKind::ConnectionReset
-                        | io::ErrorKind::ConnectionAborted => {
-                            return backoff::Error::transient(e.into());
-                        }
-                        _ => (),
-                    }
-                }
-
-                backoff::Error::permanent(e.into())
-            })
-        },
-    )
-    .await
+    { || connect(db_url) }
+        .retry(
+            ExponentialBuilder::default().with_max_delay(Duration::from_secs(opts.connect_timeout)),
```

Comment:
For example, here's another problem: `with_max_delay` is _not_ equivalent to `with_max_elapsed_time` from the old code.

Because the `backon` crate limits the _number_ of attempts, not the total duration, it makes `connect_timeout` somewhat meaningless without _also_ wrapping this block in, e.g. `tokio::time::timeout()`. That's just unnecessarily annoying.